TimberWinR/Outputs/Redis.cs

```diff
@@ -198,10 +228,16 @@ private void RedisSender()
                         try
                         {
                             string[] messages;
+                            // Exclusively
                             lock (_locker)
-                            {
-                                messages = _jsonQueue.Take(_batchCount).ToArray();
+                            {           
+                                // Take a sample of the queue depth
+                                if (_sampleCountIndex >= QUEUE_SAMPLE_SIZE)
+                                    _sampleCountIndex = 0;     
+                                _sampleQueueDepths[_sampleCountIndex++] = _jsonQueue.Count;                                                                                    
```

Comment:
Is there anything built into C# that will intelligently accumulate statistics and only store the summaries? I looked around quickly, but didn't see anything.